src/main/java/frc/robot/RobotContainer.java

```diff
@@ -183,12 +188,25 @@ private void configureBindings() {
     copilotJoystick.leftTrigger().whileTrue(m_lifter.moveToCommand(RobotState.L3_CORAL));
     copilotJoystick.rightBumper().whileTrue(m_lifter.moveToCommand(RobotState.L2_CORAL));
     copilotJoystick.leftBumper().whileTrue(m_lifter.moveToCommand(RobotState.L1_CORAL));
+    
 
+    // TEMP; two different conditions for automatic intake stop
+      
+    // stop collar after coral is seen and a delay in seconds
+    // copilotJoystick.b().whileTrue(
+    //   Commands.sequence(
+    //     m_collar.runCollar().until(m_rampSensor.coralSeenAfterRamp()),
+    //     m_collar.runCollar().withTimeout(CollarConstants.INTAKE_STOP_OFFSET),
+    //     m_collar.runCollarOff())); 
 
+    // stop collar after coral passes entirely through the collar
     copilotJoystick.b().whileTrue(
-      m_collar.runCollar()); 
+      Commands.sequence(
```

Comment:
Can we create a factory function on the collar subsystem that returns this command, so it can be used in autonomous sequences.